learn/cookbooks/langchain.mdx

```diff
@@ -0,0 +1,187 @@
+---
+title: Building semantic search with LangChain
+description: Learn how to use LangChain with Meilisearch to build semantic search using similarity search.
+---
+
+# Building semantic search with LangChain
+
+Vector search allows you to find documents that share similar characteristics. In this tutorial, we’ll use OpenAI’s text embeddings to measure the similarity between document properties. Then, we’ll use the LangChain framework to seamlessly integrate Meilisearch and build semantic search.
+
+<Capsule intent="tip">
+
+This guide uses Meilisearch’s Python SDK. But you can use Meilisearch with almost any language.
+
+</Capsule>
+
+## Requirements
+
+This tutorial assumes a basic understanding of Python and LangChain. Beginners to LangChain will still find the tutorial accessible.
+
+- Python (LangChain requires >= 3.8.1 and < 4.0) and the pip CLI
+- A Meilisearch >= 1.6 instance running — see our [quick start](/learn/getting_started/cloud_quick_start)
+- An OpenAI API key — [get yours](https://platform.openai.com/account/api-keys)
+
+## Setting up a Meilisearch instance
+
+After creating your Meilisearch instance, make sure to grab your Meilisearch host and an API key with write permissions.
+
+<Capsule intent="warning">
+
+In production, we recommend using different API keys holding only the minimal permissions to operate.
+
+</Capsule>
+
+
+Enable the vector store feature. On Meilisearch Cloud, enable it via your project’s Settings page. For self-hosted Meilisearch, activate the vector store with the [API route](/learn/experimental/vector_search#activate-vector-search).
+
+## Creating the project
+
+Create a folder for our project with an empty `main.py` file for your code.
+
+Before writing the code, install the necessary dependencies:
+
+```bash
+pip install langchain openai meilisearch python-dotenv
+```
+
+<Capsule intent="tip">
+
+This guide uses python-dotenv to load environment variables from a `.env` file. Feel free to load them in any other convenient way for you.
+
+</Capsule>
+
+If using dotenv, first create a .env to store our credentials:
```

Comment:
The user already installed dotenv, so you can skip the "If" and just give the instructions.